javaslang/src/main/java/javaslang/collection/Collections.java

```diff
@@ -22,6 +22,78 @@
  */
 final class Collections {
 
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    static boolean equals(Map<?, ?> map1, Object object) {
+        if (map1 == null ^ object == null) {
```

Comment:
As it is static method we could pass for example ```equals(null, TreeSet.of(1, 2, 3))``` that's why we need to check explicitly if none of parameters are null.